python/kfserving/README.md

```diff
@@ -1 +1,39 @@
-TODO
\ No newline at end of file
+# KFServing
+
+KFServing is a unit of model serving, and every model type (XGBosst, PyTorch, Tensorflow etc.) extends KFServing. At a highlevel it defines a model, a server and storage mechanism for models.
+
+A model has essentially three functions:
+
+* Pre processing
+* Prediction 
+* Post processing
+
+A server on the other hand provides many funtionalities, inclding among others
+
+* Registering a model and starting the server
+* Prediction Handler
+* Liveness Handler 
+* Metrics Handler 
+
+Models to load in KFServing can come from following three types of storage locations currently
+
+* Google Cloud Storage with a prefix: "gs://"
+* S3 Compatible Object Storage with a prefix "s3://"
+* Local filesystem with a prefix "/"
+
+To start the server locally on your machine for development needs, run the following command under this folder
+
+```
+pip3 install -e .
+```
+
+You would see an output similar to the one below, indicating that kfserving has been installed.
```

Comment:
The following output indicates a successful install.

Reply:
Done